swift/Samples/AlertContainer/Sources/AlertContainerViewController.swift

```diff
@@ -0,0 +1,258 @@
+import UIKit
+import Workflow
+import WorkflowUI
+
+internal final class AlertContainerViewController<AlertScreen: Screen>: ScreenViewController<AlertContainerScreen<AlertScreen>> {
+    private var baseScreenViewController: DescribedViewController
+    
+    private let dimmingView = UIView()
+    
+    private var alertView: AlertView?
+    
+    required init(screen: AlertContainerScreen<AlertScreen>, environment: ViewEnvironment) {
+        baseScreenViewController = DescribedViewController(screen: screen.baseScreen, environment: environment)
+        super.init(screen: screen, environment: environment)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        addChild(baseScreenViewController)
+        view.addSubview(baseScreenViewController.view)
+        baseScreenViewController.didMove(toParent: self)
+        
+        dimmingView.backgroundColor = UIColor(white: 0, alpha: 0.5)
+        view.addSubview(dimmingView)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        baseScreenViewController.view.frame = view.bounds
+        
+        dimmingView.frame = view.bounds
+        dimmingView.isUserInteractionEnabled = (alertView != nil)
+        dimmingView.alpha = (alertView != nil) ? 1 : 0
+    }
+    
+    override func screenDidChange(from previousScreen: AlertContainerScreen<AlertScreen>, previousEnvironment: ViewEnvironment) {
+        update()
+    }
+    
+    func update() {
+        baseScreenViewController.update(screen: screen.baseScreen, environment: environment)
+        
+        if let alert = screen.alert {
+            if let alertView = alertView {
+                alertView.alert = alert
+            }
+            else {
+                let inAlertView = AlertView(alert: alert)
+                inAlertView.backgroundColor = .init(white: 0.95, alpha: 1)
+                inAlertView.layer.cornerRadius = 10
+                inAlertView.translatesAutoresizingMaskIntoConstraints = false
+                alertView = inAlertView
+                inAlertView.accessibilityViewIsModal = true
+                view.insertSubview(inAlertView, aboveSubview: dimmingView)
+                
+                let alertWidth = CGFloat(343.0)
+                
+                var constraints: [NSLayoutConstraint] = []
+                
+                constraints.append(inAlertView.centerXAnchor.constraint(equalTo: view.centerXAnchor))
+                constraints.append(inAlertView.centerYAnchor.constraint(equalTo: view.centerYAnchor))
+                constraints.append(inAlertView.heightAnchor.constraint(greaterThanOrEqualToConstant: 0))
+                constraints.append(inAlertView.widthAnchor.constraint(greaterThanOrEqualToConstant: alertWidth))
```

Comment:
nit: can we make this a let?
```suggestion
                let constraints: [NSLayoutConstraint] = [
                    inAlertView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
                    inAlertView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
                    inAlertView.heightAnchor.constraint(greaterThanOrEqualToConstant: 0)
                    inAlertView.widthAnchor.constraint(greaterThanOrEqualToConstant: alertWidth)
                ]
```

Reply:
Further, I think we could use `NSLayoutConstraint.activate()`, and then snip the `view.addConstraints` below.

```
NSLayoutConstraint.activate([
                    inAlertView.centerXAnchor.constraint(equalTo: view.centerXAnchor)
                    inAlertView.centerYAnchor.constraint(equalTo: view.centerYAnchor)
                    inAlertView.heightAnchor.constraint(greaterThanOrEqualToConstant: 0)
                    inAlertView.widthAnchor.constraint(greaterThanOrEqualToConstant: alertWidth)
                ])
```